controllers/clusterresources/azurevnetpeering_controller_test.go

```diff
@@ -21,51 +21,79 @@ import (
-				DataCentreID: "375e4d1c-2f77-4d02-a6f2-1af617ff2ab2",
-				PeerSubnets:  []string{"172.31.0.0/16", "192.168.0.0/16"},
-			},
-			PeerResourceGroup:      "rg-1231212",
-			PeerSubscriptionID:     "sg-123321",
-			PeerADObjectID:         "ad-132313",
-			PeerVirtualNetworkName: "vnet-123123123",
-		}
-
-		ctx := context.Background()
-		resource := v1beta1.AzureVNetPeering{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "azurevnetpeering",
-				Namespace: "default",
-				Annotations: map[string]string{
-					models.ResourceStateAnnotation: models.CreatingEvent,
-				},
+var _ = Describe("AzureVNetPeering controller", func() {
+	azureVNetPeeringSpec := v1beta1.AzureVNetPeeringSpec{
+		VPCPeeringSpec: v1beta1.VPCPeeringSpec{
+			DataCentreID: "375e4d1c-2f77-4d02-a6f2-1af617ff2ab2",
+			PeerSubnets:  []string{"172.31.0.0/16", "192.168.0.0/16"},
+		},
+		PeerResourceGroup:      "rg-1231212",
+		PeerSubscriptionID:     "sg-123321",
+		PeerADObjectID:         "ad-132313",
+		PeerVirtualNetworkName: "vnet-123123123",
+	}
+
+	ctx := context.Background()
+	resource := v1beta1.AzureVNetPeering{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "azurevnetpeering",
+			Namespace: "default",
+			Annotations: map[string]string{
+				"instaclustr.com/test": "true",
 			},
-			Spec: azureVNetPeeringSpec,
-		}
+		},
+		Spec: azureVNetPeeringSpec,
+	}
+
+	key := client.ObjectKeyFromObject(&resource)
+	var doneCh chan struct{}
 
+	Context("When setting up a Azure VNet Peering CRD", func() {
 		It("Should create a Azure VNet Peering resources", func() {
 			Expect(k8sClient.Create(ctx, &resource)).Should(Succeed())
 
+			doneCh = tests.NewChannelWithTimeout(timeout)
+
 			By("Sending Azure VNet Peering Specification to Instaclustr API v2")
-			var azureVNetPeering v1beta1.AzureVNetPeering
 			Eventually(func() bool {
-				if err := k8sClient.Get(ctx, types.NamespacedName{Name: "azurevnetpeering", Namespace: "default"}, &azureVNetPeering); err != nil {
+				if err := k8sClient.Get(ctx, key, &resource); err != nil {
 					return false
 				}
 
-				return azureVNetPeering.Status.ID == mock.StatusID
+				return resource.Status.ID != ""
 			}).Should(BeTrue())
+			doneCh <- struct{}{}
+		})
+	})
+
+	Context("When deleting AzureVNetPeering k8s resource", func() {
+		It("Should delete the resource in k8s and API", func() {
+			<-doneCh
+			id := resource.Status.ID
+
+			By("sending delete request to k8s")
+			Expect(k8sClient.Delete(ctx, &resource)).Should(Succeed())
+
+			By("checking if resource exists on the Instaclustr API")
+			Eventually(func() error {
+				_, err := instaClient.GetPeeringStatus(id, instaclustr.AzurePeeringEndpoint)
+				return err
+			}, timeout, interval).Should(Equal(instaclustr.NotFound))
+
+			By("checking if the resource exists in k8s")
+			Eventually(func() bool {
+				var resource v1beta1.AWSEncryptionKey
+				return errors.IsNotFound(k8sClient.Get(ctx, key, &resource))
+			}, timeout, interval).Should(BeTrue())
```

Comment:
Why did you check for `AWSEncryptionKey` In all your tests?

Reply:
Oops, I've mistyped.

Reply:
Fixed